backend/src/main/java/com/sim_backend/websockets/observers/BootNotificationObserver.java

```diff
@@ -0,0 +1,74 @@
+package com.sim_backend.websockets.observers;
+
+import com.sim_backend.state.IllegalStateException;
+import com.sim_backend.state.SimulatorState;
+import com.sim_backend.state.SimulatorStateMachine;
+import com.sim_backend.websockets.MessageScheduler;
+import com.sim_backend.websockets.OCPPWebSocketClient;
+import com.sim_backend.websockets.events.OnOCPPMessage;
+import com.sim_backend.websockets.events.OnOCPPMessageListener;
+import com.sim_backend.websockets.messages.BootNotification;
+import com.sim_backend.websockets.messages.BootNotificationResponse;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Observer that handles BootNotification requests and responses as part of the OCPP protocol
+ * communication with the Central System.
+ */
+public class BootNotificationObserver implements OnOCPPMessageListener {
```

Comment:
The BootNotificationObserver also needs to subscribe to state changes. When we move into a BootingUp state, it should automatically send the BootNotification message.